### PR TITLE
Add videofeed docker folder

### PR DESCRIPTION
**Pull request by gdynusa**

### CODE REVIEW

**Review by petrbel — APPROVED** [review]
I believe this is exactly what we were looking for. Nevertheless, @FloopCZ should provide an independent review :)

File: dockerfiles/videofeed/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+WORKDIR /app
+
+# Install OpenCV
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends python3-opencv python3-pip \
+ && rm -rf /var/lib/apt/lists/*
+
+# Install python requirements
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
```
Comment (FloopCZ) [inline dockerfiles/videofeed/Dockerfile:15]:
```suggestion
RUN pip3 install --no-cache-dir -r requirements.txt
```

File: dockerfiles/videofeed/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Video Feed
+Docker image providing video feed from device on port 5000
```
Comment (FloopCZ) [inline dockerfiles/videofeed/README.md:2]:
```suggestion
Docker image providing video feed from V4L device on port 5000.
```

File: dockerfiles/videofeed/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+## Build Image
+```docker build -t videofeed .```
```
Comment (FloopCZ) [inline dockerfiles/videofeed/README.md:5]: @petrbel To have it consistent with the rest of the images, consider:
```suggestion
```docker build -t iterait/videofeed:latest .```
```

File: dockerfiles/videofeed/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+## Run container
+Run video feed on port 1234 providing data from camera /dev/video0. Make sure to specify env variable DEVICE to match camera.
+
+```docker run --rm -d --device=/dev/video0 -e DEVICE=/dev/video0 -p 1234:5000 videofeed```
```
Comment (FloopCZ) [inline dockerfiles/videofeed/README.md:10]:
```suggestion
```docker run --rm -d --device=/dev/video0 -e DEVICE=/dev/video0 -p 1234:5000 iterait/videofeed```
```

File: dockerfiles/videofeed/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Video Feed
+Docker image providing video feed from device on port 5000
+
+## Build Image
+```docker build -t videofeed .```
+
+## Run container
+Run video feed on port 1234 providing data from camera /dev/video0. Make sure to specify env variable DEVICE to match camera.
```
Comment (FloopCZ) [inline dockerfiles/videofeed/README.md:8]:
```suggestion
Run video feed on port 1234 providing data from camera /dev/video0. Make sure to specify env variable DEVICE to match the camera device.
```

File: dockerfiles/videofeed/video_feed.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class VideoCamera:
+    def __init__(self):
+        self._device = os.environ.get('DEVICE', None)
+        if self._device is None:
+            raise ValueError(f'Required env. variable `DEVICE` not set.')
+        self._video = cv2.VideoCapture(self._device)
+
+    def __del__(self):
```
Comment (FloopCZ) [inline dockerfiles/videofeed/video_feed.py:14]: This class should be implemented as a resource instead.

**Review by FloopCZ — COMMENTED** [review]
Thank you @petrbel and @gdynusa. Although I am not sure this is the right way to do it see the comments in the corresponding `VideoReader` pull request. :-)

**Pull request closed without merging**